runtime/vm/ClassInitialization.cpp

```diff
@@ -391,6 +391,9 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 					clazz = VM_VMHelpers::currentClass(clazz);
 				}
 				VM_ObjectMonitor::exitObjectMonitor(currentThread, initializationLock);
+				/*Initialize a Flattened Class Cache */
+				j9object_t classReferenceObject = vm->memoryManagerFunctions->J9AllocateObject(currentThread, clazz, J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
```

Comment:
need to perform a NULL check after the allocation to determine if the object was allocated. If it returns null need to setheapOOM. You should be able to find examples of this in other places where `J9AllocateObject` is used